### PR TITLE
[FEATURE] Add system notification

### PR DESCRIPTION
**Pull request by serhiislobodian**

### CODE REVIEW

**Comment by lverba** [comment]
retest this please

File: frontend/components/modal/system-notifications.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    };
+
+    resolveNotifications = () => {
+        const { dispatch } = this.props;
+        analytics.event({
+            action: "System Notifications Modal",
+            category: "Modal Windows",
+            label: "Enable notifications",
+        });
+        dispatch(accountOperations.setSystemNotificationsStatus(6));
```
Comment (boodyvo) [inline frontend/components/modal/system-notifications.js]: What is 6? Create enum for this

File: frontend/components/modal/system-notifications.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+    rejectNotifications = () => {
+        const { dispatch } = this.props;
+        if (this.state.showAgain) {
+            analytics.event({
+                action: "System Notifications Modal",
+                category: "Modal Windows",
+                label: "Disable notifications",
+            });
+            dispatch(accountOperations.setSystemNotificationsStatus(3));
```
Comment (boodyvo) [inline frontend/components/modal/system-notifications.js]: Also enum

File: frontend/components/modal/system-notifications.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        } else {
+            analytics.event({
+                action: "System Notifications Modal",
+                category: "Modal Windows",
+                label: "Disable notifications, never ask again",
+            });
+            dispatch(accountOperations.setSystemNotificationsStatus(2));
```
Comment (boodyvo) [inline frontend/components/modal/system-notifications.js]: Also enum

File: frontend/modules/account/reducers.js [hunk 4]
```diff
@@ -30,6 +30,7 @@ export const initStateAccount = {
     },
     peers: [],
     signedMessage: null,
+    systemNotifications: types.DISABLED,
```
Comment (boodyvo) [inline frontend/modules/account/reducers.js]: Remove extra type

File: frontend/modules/channels/operations.js [hunk 5]
```diff
@@ -50,24 +50,108 @@ function getChannels() {
                 const bChan = b.chan_id;
                 return aChan < bChan ? -1 : 1;
             };
+            const keyByListChannels = keyBy(response.response.channels, channel => channel.channel_point.split(":")[0]);
+            Object.values(dbChannels).forEach((dbChan) => {
+                if (
+                    dbChan.status === "active"
+                    && !has(keyByListChannels, dbChan.fundingTxid)
+                    && getState().channels.deleteQueue.filter(channel =>
```
Comment (boodyvo) [inline frontend/modules/channels/operations.js]: add a comment here like "Check if the channel is not closing and active"

File: frontend/modules/channels/operations.js [hunk 6]
```diff
@@ -50,24 +50,108 @@ function getChannels() {
...
+                            } else {
+                                dispatch(appOperations.sendSystemNotification({
+                                    body: "Channel becomes inactive",
+                                    title: chanName,
+                                }));
+                            }
+                        }
+
+                        if (
+                            dbChan.localBalance / totalBalance > 0.1
```
Comment (boodyvo) [inline frontend/modules/channels/operations.js]: Put const = 10(%), not value.

File: frontend/modules/channels/operations.js [hunk 7]
```diff
@@ -98,20 +182,38 @@ function getChannels() {
                 if (has(dbChannels, chanTxid)) {
                     const dbChan = dbChannels[chanTxid];
                     chanName = dbChan.name;
-                    // maturity = blockHeight - dbChannels[channel.channel.channel_point].blockHeight;
                     maturity = getState().onchain.history
-                        .filter(txn => txn.tx_hash === dbChan.fundingTxid.split(":")[0])
+                        .filter(txn => txn.tx_hash === dbChan.fundingTxid)
```
Comment (boodyvo) [inline frontend/modules/channels/operations.js]: Add comment

File: frontend/modules/onchain/operations.js [hunk 8]
```diff
@@ -65,6 +64,69 @@ function getOnchainHistory() {
                     blockHash = chainTxns[txn].block_hash;
                     blockHeight = chainTxns[txn].block_height;
                     totalFees = parseInt(chainTxns[txn].total_fees, 10);
+                    if (!has(dbTxns, txn)) {
+                        if (amount > 0 && numConfirmations >= 1) {
```
Comment (boodyvo) [inline frontend/modules/onchain/operations.js]: Make 1 as a constant

File: frontend/modules/onchain/operations.js [hunk 9]
```diff
@@ -65,6 +64,69 @@ function getOnchainHistory() {
...
+                    } else if (
+                        dbTxns[txn].amount !== amount
+                        || dbTxns[txn].blockHeight !== blockHeight
+                        || dbTxns[txn].blockHash !== blockHash
+                        || dbTxns[txn].status !== status
+                        || dbTxns[txn].timeStamp !== chainTxns[txn].time_stamp
+                        || dbTxns[txn].totalFees !== totalFees
+                        || dbTxns[txn].name === ""
+                        || (
+                            dbTxns[txn].numConfirmations < 6
```
Comment (boodyvo) [inline frontend/modules/onchain/operations.js]: 6 is also const

**Pull request merged**